File: apple/Sources/ReactNativeWatchOS/Modifiers/TabItem.swift
import SwiftUI

/// SwiftUI `.tabItem { … }` (watchOS 7+). Sets the tab-bar label for a view
/// used as a page inside a `TabView`. The `content` slot is resolved via
/// `ctx`; with no content provided the view is returned unchanged.
enum RNWTabItemModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("tabItem") { view, params, ctx in
            guard let body = ctx.content(params.string("content")) else {
                return view
            }
            return AnyView(view.tabItem { body })
        }
    }
}

File: src/modifiers/tabItem.ts
import type { ReactNode } from 'react';
import { createModifier } from './createModifier';

export interface TabItemParams {
  /// The tab's label — typically a `Label` or `Text`/`Image` — shown in the
  /// `TabView`'s tab bar.
  content: ReactNode;
}

/// SwiftUI `.tabItem { … }` (watchOS 7+). Sets the label for a view used as a
/// page inside a `TabView`.
export function tabItem(content: ReactNode): ReturnType<typeof createModifier>;
export function tabItem(params: TabItemParams): ReturnType<typeof createModifier>;
export function tabItem(a: TabItemParams | ReactNode) {
  if (a && typeof a === 'object' && 'content' in a) {
    return createModifier('tabItem', a as TabItemParams);
  }
  return createModifier('tabItem', { content: a as ReactNode });
}
